src/services/tasks.service.ts

```diff
@@ -8,4 +9,70 @@ export const getTaskByStatus = async (status: string): Promise<Task[]> => {
     return await Task.findAll({
         where: { status }
     });
+}
+
+export const getTaskByID = async (id: string): Promise<Task> => {
+    try{
+        const task = await Task.findOne({
+            where: { id }
+        });
+
+        if(task !== null){
+            return task;
+        }else{
+            throw('Could not find task ID in database');
+        }   
+    }catch(err){
+        return(err);
+    }
+}
+
+export const addTask = async (task: string): Promise<string> => {
+    const newTask = await Task.create({task: task, status: ETaskStatus.Incomplete});
+    return(newTask.id);
+}
+
+export const deleteTaskByID = async (id: string): Promise<void> => {
+    try{
+        const task = await getTaskByID(id);
+    
+        if(task.id === undefined){
+            throw(task);
```

Comment:
We should not throw `task` but the error. 

```suggestion
            throw new Error(`Task ${task.id} is not existed`)
```

Also can use this package to add more data into the error object https://www.npmjs.com/package/err-code